akka-cluster-sharding/src/main/scala/akka/cluster/sharding/external/javadsl/ExternalShardAllocationClient.scala

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright (C) 2019-2020 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.cluster.sharding.external.javadsl
+
+import java.util.concurrent.CompletionStage
+
+import akka.Done
+import akka.actor.Address
+import akka.annotation.ApiMayChange
+import akka.cluster.sharding.ShardRegion.ShardId
+import akka.cluster.sharding.external.ShardLocations
+
+/**
+ * API May Change
+ */
```

Comment:
Perhaps also "Not for user extension" just to make it clear it is a trait for consumption, not extension

Reply:
:+1: